Update CHANGELOG.md for v1.10.0 (#2479)
bottlerocket-os/bottlerocket: CHANGELOG.md

# v1.10.0 (2022-10-10)

## OS Changes

> * Automated reconciliation of boot settings ([#2375])

Reviewer: ```suggestion
* Add optional setting to reboot into new kernel command line parameters ([#2375])
```

* Support for static IP addressing ([#2204], [#2330], [#2445])

> * Add sources for NVIDIA's 515 driver ([#2455])

Reviewer: Can you find a better way to phrase this? We certainly don't have access to NVIDIA's sources.

* Set mode for tmpfs mounts ([#2473])
* Increase inotify default limits ([#2335])

> * Increase `vm.max_map_count` to match AL2's configurations ([#2344])

Reviewer: maybe something like:
```suggestion
* Align `vm.max_map_count` with the EKS Optimized AMI ([#2344])
```

> * Add support to configure DNS settings through the API ([#2353])

Reviewer 2: This might read better as something like:
"Add support for configuring DNS settings..."

* Migrate `netdog` from `serde_xml_rs` to `quick-xml` ([#2311])
* Support versioning for `net.toml` ([#2281])
* Update admin and control container ([#2471], [#2472])

## Orchestrator Changes

### Kubernetes

* Add support for Kubernetes 1.24 variants ([#2437])

> * Remove variant aws-k8s-1.19 ([#2316])

Reviewer: ```suggestion
* Remove Kubernetes 1.19 variants ([#2316])
```

I know there was just the one, but for consistency with future add/remove news.

* Increase the kube-api-server QPS from 5/10 to 10/20 ([#2436], thanks @tzneal)
* Update eni-max-pods with new instance types ([#2416])
* Add setting to change `kubelet`'s log level ([#2460], [#2470])

## Platform Changes

### AWS
* Disable drivers for USB-attached network interfaces ([#2328])

### Metal
* Add driver support for Solarflare, Pensando, Myricom, Huawei, Emulex, Chelsio, Broadcom, AMD and Intel 10G+ network cards ([#2379])

## Build Changes
* Extend `external-files` to vendor go modules ([#2378], [#2403], [#2430])
* Make `net_config` unit tests reusable across versions ([#2385])

> * Add diff-kernel-config to identify kernel config changes ([#2368])

Reviewer: ```suggestion
* Add `diff-kernel-config` to identify kernel config changes ([#2368])
```

* Extended support for variants in buildsys ([#2339])
* Clarify crossbeam license ([#2447])

> * Switch back to `simplelog` in `pubsys` ([#2454])

Reviewer: Let's roll this into a changelog item about moving pubsys from Rusoto to AWS SDK for Rust.

* Honor `BUILDSYS_ARCH` and `BUILDSYS_VARIANT` env variables when set ([#2425])
* Use architecture specific json payloads in unit tests ([#2367], [#2363])
* Add unified `check` target in `Makefile.toml` for review readiness ([#2384])
* Add `cargo make` tasks for testing ECS variants ([#2348])

> * Add `cargo make` tasks to perform migration testing in AWS K8s ([#2273])

Reviewer: I'd like to move this under the Kubernetes news, and the ECS ones to the ECS section.

```suggestion
* Add `cargo make` tasks to perform migration testing for Kubernetes variants in AWS ([#2273])
```

> * Update Go dependencies of fist-party go projects ([#2424], [#2440], [#2450], [#2452], [#2456])

Reviewer: ```suggestion
* Update Go dependencies of first-party go projects ([#2424], [#2440], [#2450], [#2452], [#2456])
```

👊🏻

* Update Rust dependencies ([#2458])
* Update third-party packages ([#2397], [#2398], [#2464], [#2465], thanks @kschumy)
* Update Bottlerocket SDK to 0.27.0 ([#2428])

> * Update AWS SDK Rust ([#2415], [#2476])

Reviewer: ```suggestion
* Migrate `pubsys` to the AWS SDK for Rust ([#2415], [#2476])
```

#2476 can go into the "Update Rust dependencies" line.

* Update `testsys` dependencies ([#2392])
* Fix `hotdog`'s spec URL to the correct upstream link ([#2326])
* Fix clippy warnings and enable lints on pull requests ([#2337], [#2346], [#2443])
* Format issue field in PR template ([#2314])

## Documentation Changes
* Update checksum for new `root.json` ([#2405])

> * Mention that boot settings are available in k8s-1.23 variants ([#2358])

Reviewer: ```suggestion
* Mention that boot settings are available in Kubernetes 1.23 variants ([#2358])
```

> * Mention the need of AWS credentials in BUILDING.md and PUBLISHING-AWS.md ([#2334])

Reviewer: ```suggestion
* Mention the need for AWS credentials in BUILDING.md and PUBLISHING-AWS.md ([#2334])
```

* Add China to supported regions lists ([#2315])
* Add community section to README.md ([#2305], [#2383])
* Standardize `userdata.toml` as the filename used in different docs ([#2446])
* Remove commit from image name in PROVISIONING-METAL.md ([#2312])
* Add note to CONTRIBUTING.md that outlines filenames' casing ([#2306])

> * Fix typos in `Makefile.toml`, QUICKSTART-ECS.md, QUICKSTART-EKS.md, `netdog` and `prairiedog` ([#2318])

Reviewer: ```suggestion
* Fix typos in `Makefile.toml`, QUICKSTART-ECS.md, QUICKSTART-EKS.md, `netdog` and `prairiedog` ([#2318], thanks @kianmeng)
```

* Fix casing for GitHub and VMware in CHANGELOG.md ([#2329])
* Fix typo in test setup command ([#2477])
* Fix TESTING.md link typo ([#2438])
* Fix positional `fetch-license` argument ([#2457])

[#2375]: https://github.com/bottlerocket-os/bottlerocket/pull/2375
[#2204]: https://github.com/bottlerocket-os/bottlerocket/pull/2204
[#2330]: https://github.com/bottlerocket-os/bottlerocket/pull/2330

> [#2445]: https://github.com/bottlerocket-os/bottlerocket/pull/2445

Reviewer: nit: I would recommend sorting this list so it's easier to scan visually for duplicates.

# v1.10.0 (2022-10-10)

## OS Changes
* Add optional settings to reboot into new kernel command line parameters ([#2375])
* Support for static IP addressing ([#2204], [#2330], [#2445])
* Add support for NVIDIA driver version 515 ([#2455])
* Set mode for tmpfs mounts ([#2473])
* Increase inotify default limits ([#2335])
* Align `vm.max_map_count` with the EKS Optimized AMI ([#2344])
* Add support for configuring DNS settings ([#2353])
* Migrate `netdog` from `serde_xml_rs` to `quick-xml` ([#2311])
* Support versioning for `net.toml` ([#2281])
* Update admin and control container ([#2471], [#2472])

## Orchestrator Changes

### ECS
* Add `cargo make` tasks for testing ECS variants ([#2348])

### Kubernetes

* Add support for Kubernetes 1.24 variants ([#2437])

> * Remove Kubernetes aws-k8s-1.19 variants ([#2316])

Reviewer: nit:
```suggestion
* Remove Kubernetes 1.19 variants ([#2316])
```
